utilsForFrontend.mjs

```diff
@@ -475,7 +473,14 @@ export function isVisible(fnodeOrElement) {
     if (elementStyle.visibility === 'hidden') {
         return false;
     }
+    // Check if the element is off-screen
     const frame = element.ownerDocument.defaultView;
+    if (elementRect.x + elementRect.width < 0
+        || (elementRect.y + elementRect.height) < 0
+        || (elementRect.x > frame.innerWidth || elementRect.y > frame.innerHeight)
```

Comment:
It seems like the parens around this pair of conditions shouldn't be here, since you didn't put them around the top pair (and they don't affect the logic).